### PR TITLE
Bugfix/privacy toggle

### PR DESCRIPTION
**Pull request by guaj**

### CODE REVIEW

**Review by mnashat — CHANGES_REQUESTED** [review]
Great job! I have one concern regarding the GetAuthentication() method, which is outlined in my comments. Thanks!

**Comment by mnashat** [comment]
Also, in files SignIn.js (line 40) and Calendar.js (lines 206, 208, 210, 216), please replace == with ===. IntelliJ is NOT happy. :(

File: src/components/StudyRoom/StudyRoomChat/Sockets.js [hunk 1]
```diff
@@ -1,9 +1,6 @@
 import io from "socket.io-client"
 
 const connectionOptions =  {
```
Comment (kikkiii21) [inline src/components/StudyRoom/StudyRoomChat/Sockets.js:3]: this is out of the scope of the bugfix privacy toggle

File: src/components/StudyRoom/StudyRoomHome.js [hunk 2]
```diff
@@ -11,19 +11,20 @@ import Divider from "@mui/material/Divider";
 import {useNavigate} from "react-router";
 import axios from "axios";
```
Comment (kikkiii21) [inline src/components/StudyRoom/StudyRoomHome.js:13]: this is out of the scope of the bugfix privacy toggle

File: src/components/CustomMUIComponents/CommonForms.js [hunk 3]
```diff
@@ -85,4 +105,87 @@ export const ProgramSelect = ({userData, handleProgramChange}) => (
...
+    padding: 3,
+    '& .MuiSwitch-switchBase': {
+        padding: 0,
+        margin: 2,
+        transitionDuration: '300ms',
+        color: '#C8C8C8',
+        '&.Mui-checked': {
+            transform: 'translateX(16px)',
+            color: '#C8C8C8',
+            '& + .MuiSwitch-track': {
```
Comment (kikkiii21) [inline src/components/CustomMUIComponents/CommonForms.js:155]: I would just put colors in our theme for the toggle 
#912338 for the red and use #C8C8C8 everywhere where there is green just to be consistent

**Review by kikkiii21 — CHANGES_REQUESTED** [review]
for the sake of consistency I would update the colors for the toggle as per my comment

File: src/components/StudyRoom/StudyRoomChat/Sockets.js (hunk 1 above)
Comment (guaj) [reply src/components/StudyRoom/StudyRoomChat/Sockets.js:3]: its a security issue. Was calling undocumented props.

**Review by mnashat — APPROVED** [review]
Great job! Approved.

**Review by kikkiii21 — APPROVED** [review]

**Pull request merged**